neqo-crypto/src/agent.rs

```diff
@@ -768,11 +766,8 @@ struct ZeroRttCheckState {
 }
 
 impl ZeroRttCheckState {
-    pub fn new(
-        fd: *mut ssl::PRFileDesc,
-        checker: Box<dyn ZeroRttChecker>,
-    ) -> Box<ZeroRttCheckState> {
-        Box::new(ZeroRttCheckState { fd, checker })
+    pub fn new(fd: *mut ssl::PRFileDesc, checker: Box<dyn ZeroRttChecker>) -> Box<Self> {
```

Comment:
I seem to remember clippy getting snippy about not returning `Self` from functions called `new`.

Reply:
Hmm It didnt trigger _yet_, maybe it's in clippy::Restriction ?